### DISCUSSION TITLE
Loading multi-modal predictor causes latency on "cold-start"

### DISCUSSION THREAD

**Discussion by rh-ia** [autogluon/autogluon#3992] [Q&A]
Hi all,

# Problem description + context
Using the autogluon MultiModal predictor, we experience some latency loading it with pickle.load() where we pickled the MultiModal predictor as follows: 

```
pickle_file = "model.pkl"

# Dumping model pickle
saved_model = {
    "model": multi_modal_predictor,  # trained predictor
    "y_col": args.label_column,
    "hyperparams": hyperparams,
}

pickle.dump(saved_model, open(pickle_file, "wb"))
```
Training with:
```
  | Name              | Type             | Params
-------------------------------------------------------
0 | model             | NumericalMLP     | 76.7 K
1 | validation_metric | MeanSquaredError | 0     
2 | loss_func         | MSELoss          | 0     
-------------------------------------------------------
```

Now we see this in the logs of our inference compute instance:
```
Created a temporary directory at /tmp/tmpffiwam1q
Writing /tmp/tmpffiwam1q/_remote_module_non_scriptable.py
```

- **After creation of this directory, the loading is 100x faster (0.5 seconds vs 50 seconds)**
- This does not happen training with a TabularPredictor.

## OS / Environment

- Python 3.8
- Autogluon 1.0.0

Inference compute (Azure function) is on Debian GNU/Linux 11 (bullseye) with the following specs:
```
Architecture:                       x86_64
CPU op-mode(s):                     32-bit, 64-bit
Byte Order:                         Little Endian
Address sizes:                      48 bits physical, 48 bits virtual
CPU(s):                             2
On-line CPU(s) list:                0,1
Thread(s) per core:                 2
Core(s) per socket:                 1
Socket(s):                          1
NUMA node(s):                       1
Vendor ID:                          AuthenticAMD
CPU family:                         25
Model:                              1
Model name:                         AMD EPYC 7763 64-Core Processor
Stepping:                           1
CPU MHz:                            3114.749
BogoMIPS:                           4890.85
Virtualization:                     AMD-V
Hypervisor vendor:                  Microsoft
Virtualization type:                full
L1d cache:                          32 KiB
L1i cache:                          32 KiB
L2 cache:                           512 KiB
L3 cache:                           32 MiB
NUMA node0 CPU(s):                  0,1
```

The Azure Function is set on the Premiun plan with "**Always On**" of at least two instance, so colds start shouldn't be the issue here.

For training we used an azure  [NCv3 series](https://learn.microsoft.com/en-us/azure/virtual-machines/ncv3-series) (1x Tesla V100 GPU)

# Question

- How can we avoid the tmp file creation ("cold start"), which seems to lead to latency and originates from pytorch?

**Thanks ahead for any suggestion!**

**Comment by trenchantai** [comment]
I think I'm having a similar issue with Timeseries Predictor - the cold start, as you put it, is very slow but subsequent runs are fast. 
Did you happen to find a solution or workaround? 
Thanks!
